crates/noirc_evaluator/src/brillig/memory.rs

```diff
@@ -0,0 +1,15 @@
+/// Simple memory allocator for brillig
+/// For now it just tracks a free memory pointer
```

Comment:
As for when we 'get smarter', my vote is only get smarter once we have concrete numbers from a public VM implementation